src/frontend/planner_test/tests/testdata/index_selection.yaml

```diff
@@ -60,6 +52,51 @@
     BatchExchange { order: [], dist: Single }
       BatchLookupJoin { type: Inner, predicate: idx2.t1._row_id IS NOT DISTINCT FROM t1._row_id, output: [t1.a, t1.b, t1.c] }
         BatchScan { table: idx2, columns: [idx2.t1._row_id], scan_ranges: [idx2.b = Decimal(Normalized(1))], distribution: SomeShard }
+- sql: |
+    /* Index on all required columns */
+    create table t1 (a int, b int, c int);
+    create index idx1 on t1(a, b, c);
+    select * from t1 order by a
+  batch_plan: |
+    BatchExchange { order: [idx1.a ASC], dist: Single }
+      BatchScan { table: idx1, columns: [idx1.a, idx1.b, idx1.c], distribution: UpstreamHashShard(idx1.a, idx1.b, idx1.c) }
+- sql: |
+    /* Index includes all required columns I */
+    create table t1 (a int, b int, c int);
+    create index idx1 on t1(b) include(c, a);
+    select * from t1 order by b
+  batch_plan: |
+    BatchExchange { order: [idx1.b ASC], dist: Single }
+      BatchScan { table: idx1, columns: [idx1.a, idx1.b, idx1.c], distribution: UpstreamHashShard(idx1.b) }
```

Comment:
The same setup that causes problems in the e2e test is also shown here.
Somehow the index scan does not provide the order in this case. @st1page